internal/vehicle_template.go

```diff
@@ -66,7 +66,7 @@ func (vt *vehicleTemplates) GetTemplateSettings(addr *common.Address) (*models.T
 	// if no change, just return what we have
 	if templateURLsLocal != nil &&
 		templateURLsRemote.PidURL == templateURLsLocal.PidURL &&
-		templateURLsRemote.DeviceSettingURL == templateURLsLocal.DeviceSettingURL {
+		templateURLsRemote.DeviceSettingURL == templateURLsLocal.DeviceSettingURL && deviceSettings != nil {
```

Comment:
@JamesReate , I added a check for deviceSettings here, as even if it was nil, we are entering that if and do not try to query it remotely

Reply:
yes - we should write the deviceSettings, similar to what we do with the pids and dbc, so that if the url's are the same we don't pull them again